### ISSUE TITLE
problem with dual gradient echo fieldmap

### ISSUE THREAD

**Issue by elhavee** [jmtyszka/bidskit#31]
Hi Mike,

Thanks for this toolbox!
We are having an issue with fieldmaps (Siemens dual gradient echo). We have four functional runs and one fieldmap for each. Everything seems to be working for one subject, here are the files in the work directory for one of the runs:

work/conversion/sub-pilot02/ses-first/

> de44ea9386c2e21e2a39718bc121a94d2a0ee73a--OBIWAN_FIELD_MAP--GR--12_e2.json
> de44ea9386c2e21e2a39718bc121a94d2a0ee73a--OBIWAN_FIELD_MAP--GR--12_e2.nii.gz
> de44ea9386c2e21e2a39718bc121a94d2a0ee73a--OBIWAN_FIELD_MAP--GR--12.json
> de44ea9386c2e21e2a39718bc121a94d2a0ee73a--OBIWAN_FIELD_MAP--GR--12.nii.gz
> de44ea9386c2e21e2a39718bc121a94d2a0ee73a--OBIWAN_FIELD_MAP--GR--13.json
> de44ea9386c2e21e2a39718bc121a94d2a0ee73a--OBIWAN_FIELD_MAP--GR--13.nii.gz
> de44ea9386c2e21e2a39718bc121a94d2a0ee73a--OBIWAN_RUN4--EP--14.json
> de44ea9386c2e21e2a39718bc121a94d2a0ee73a--OBIWAN_RUN4--EP--14.nii.gz
> 

And the files in the BIDS output directory for two runs:

BIDS/sub-pilot02/ses-first/fmap/
> sub-pilot02_ses-first_run-01_acq-task_phasediff.json
> sub-pilot02_ses-first_run-01_acq-task_phasediff.nii.gz
> sub-pilot02_ses-first_run-02_acq-task_magnitude.nii.gz
> sub-pilot02_ses-first_run-04_acq-task_phasediff.json
> sub-pilot02_ses-first_run-04_acq-task_phasediff.nii.gz
> sub-pilot02_ses-first_run-05_acq-task_magnitude.nii.gz

From what we can tell, the numbering is odd because a run number is being used for the echo 2 magnitude which is being discarded. 

However for another subject with the exact same DICOM file structure, we are encountering a problem. Here is the same run in the work directory after the first pass:

work/conversion/sub-pilot03/ses-first/
> 2da3451ba5c5b36e641f3f0f82ae778dde5bfb95--OBIWAN_FIELD_MAP--GR--12_e2.json
> 2da3451ba5c5b36e641f3f0f82ae778dde5bfb95--OBIWAN_FIELD_MAP--GR--12_e2.nii.gz
> 2da3451ba5c5b36e641f3f0f82ae778dde5bfb95--OBIWAN_FIELD_MAP--GR--12.json
> 2da3451ba5c5b36e641f3f0f82ae778dde5bfb95--OBIWAN_FIELD_MAP--GR--12.nii.gz
> 2da3451ba5c5b36e641f3f0f82ae778dde5bfb95--OBIWAN_FIELD_MAP--GR--13_e2.json
> 2da3451ba5c5b36e641f3f0f82ae778dde5bfb95--OBIWAN_FIELD_MAP--GR--13_e2.nii.gz
> 2da3451ba5c5b36e641f3f0f82ae778dde5bfb95--OBIWAN_RUN4--EP--14.json
> 2da3451ba5c5b36e641f3f0f82ae778dde5bfb95--OBIWAN_RUN4--EP--14.nii.gz

The series for the phasediff files is being named differently ('13_e2' instead of '13'), and at the second pass we get the following error:

```
Organizing OBIWAN_FIELD_MAP
    Identifying fieldmap image type
    GRE detected
    Identifying magnitude and phase images
Traceback (most recent call last):
  File "dcm2bids.py", line 944, in <module>
    main()
  File "dcm2bids.py", line 233, in main
    bids_run_conversion(work_conv_dir, first_pass, prot_dict, bids_src_ses_dir, SID, SES, overwrite)
  File "dcm2bids.py", line 353, in bids_run_conversion
    overwrite)
  File "dcm2bids.py", line 437, in bids_purpose_handling
    TE1, TE2 = bids_fmap_echotimes(work_json_fname)
  File "dcm2bids.py", line 758, in bids_fmap_echotimes
    mag1_ser_no = str(int(ser_no) - 1)
ValueError: invalid literal for int() with base 10: '13_e2'
```

If we manually modify the name of the series to match the first subject (removing the '_e2') in the work/conversion/ directory, the second pass works with no errors.
Do you have any idea why this is happening? Also, is there a way to have the output BIDS fmap files not skip over numbers? We are having this issue both with the newest and previous release of dcm2bids. 

Thanks,
Lavinia

**Comment by jmtyszka** [comment]
Thanks for raising this issue. You might have seen some recent updates attempting to finally get the run-* tag correct in the filename and the multiecho and multiple recon from the the GRE fieldmapping confuses the current code. Working on a fix now - expect a new commit by early next week.

**Comment by bryjack0890** [comment]
Hi, 

I'm having a similar issue using the newest version of bidskit (dcm2bids v1.1.2). I have one session worth of files: a T1w, dual gradient echo fieldmap, and some EPI sequences, all of which are "Intended-for." I used the --no-sessions flag when I ran the command. 

These are the files after the first pass:

101--gre_field_mapping--GR--2_e1.json
101--gre_field_mapping--GR--2_e1.nii.gz
101--gre_field_mapping--GR--2.json
101--gre_field_mapping--GR--2.nii.gz
101--gre_field_mapping--GR--3.json
101--gre_field_mapping--GR--3.nii.gz

After the second pass. I'm left with these:

sub-101_run-02_acq-rest_phasediff.json    
sub-101_run-03_acq-rest_magnitude.nii.gz
sub-101_run-02_acq-rest_phasediff.nii.gz

Any ideas about what I need to do to make it work? 
Bryan
